moveit_setup_assistant/moveit_setup_framework/src/urdf_config.cpp

```diff
@@ -92,32 +92,45 @@ void URDFConfig::loadFromPath(const std::filesystem::path& urdf_file_path, const
 
 void URDFConfig::setPackageName()
 {
-  bool package_found = extractPackageNameFromPath(urdf_path_, urdf_pkg_name_, urdf_pkg_relative_path_);
-  if (!package_found)
-  {
-    urdf_pkg_name_ = "";
-    urdf_pkg_relative_path_ = urdf_path_;  // just the absolute path
-  }
-  // Check that ROS can find the package
-  const std::filesystem::path robot_desc_pkg_path = getSharePath(urdf_pkg_name_);
+  // Reset to defaults: no package name, relative path is set to absolute path
+  urdf_pkg_name_ = "";
+  urdf_pkg_relative_path_ = urdf_path_;
 
-  if (robot_desc_pkg_path.empty())
+  std::string pkg_name;
+  std::filesystem::path relative_path;
+  if (extractPackageNameFromPath(urdf_path_, pkg_name, relative_path))
   {
-    RCLCPP_WARN(*logger_,
-                "Package Not Found In ROS Workspace. ROS was unable to find the package name '%s'"
-                " within the ROS workspace. This may cause issues later.",
-                urdf_pkg_name_.c_str());
+    // Check that ROS can find the package, update members accordingly
+    const std::filesystem::path robot_desc_pkg_path = getSharePath(pkg_name);
+    if (!robot_desc_pkg_path.empty())
+    {
+      urdf_pkg_name_ = pkg_name;
+      urdf_pkg_relative_path_ = relative_path;
+    }
+    else
+    {
+      RCLCPP_WARN(*logger_,
+                  "Package Not Found In ROS Workspace. ROS was unable to find the package name '%s'"
+                  " within the ROS workspace. This may cause issues later.",
+                  pkg_name.c_str());
```

Comment:
this warning seems very redundant. Can we maybe make it shorter while we're here?

```suggestion
      RCLCPP_WARN(*logger_,
                  "ROS was unable to find the package %s. This may cause issues later.",
                  pkg_name.c_str());
```